includes/admin/class-wc-rest-payments-orders-controller.php

```diff
@@ -168,7 +209,7 @@ public function capture_terminal_payment( $request ) {
 				'id'     => $intent->get_id(),
 			];
 
-			$result = $is_intent_captured ? $result_for_captured_intent : $this->gateway->capture_charge( $order, false );
+			$result = ( $is_intent_captured && $is_terminal_payment ) ? $result_for_captured_intent : $this->gateway->capture_charge( $order, false );
```

Comment:
The call to `capture_charge` is now further hidden down the ternary chain. An opportunity to refactor this ternary into an `if-else`?